rmgpy/data/base.py

```diff
@@ -875,10 +875,10 @@ def matchNodeToChild(self, parentNode, childNode):
             return False
         
         #If the parentNode is a Group and the childNode is a LogicOr there is nothing to check,
-        #so it gets an automatic pass. However, we do need to check that everything down this
+        #except that the parent is listed in the attributes. However, we do need to check that everything down this
         #family line is consistent, which is done in the databaseTest unitTest
         elif isinstance(parentNode.item, Group) and isinstance(childNode.item, LogicOr):
-            return True
+            return childNode.parent is parentNode
```

Comment:
As said in the comment above, we actually do this in [databaseTest.py](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/059b895b67370b2daa5699c3b5c84ffc87a36add/testing/databaseTest.py#L245-L253
l). 

In my opinion, the current implementation, unless we want to start using the matchNodeToChild for things other than database unittests.